src/commands/force/source/delete.ts

```diff
@@ -89,15 +101,41 @@ export class Delete extends DeployCommand {
   private tempDir = path.join(os.tmpdir(), 'source_delete');
 
   public async run(): Promise<DeployCommandResult> {
+    await this.preChecks();
     await this.delete();
+    // result.getFileResponses() will crawl the tree, but that would throw after the delete occurs.
+    // Extract them here for updateTracking to use later
+    const fileResponses = this.mixedDeployDelete.delete.length
+      ? this.mixedDeployDelete.delete
+      : this.deployResult.getFileResponses();
     await this.resolveSuccess();
     const result = this.formatResult();
     // The DeleteResultFormatter will use SDR and scan the directory, if the files have been deleted, it will throw an error
     // so we'll delete the files locally now
     await this.deleteFilesLocally();
+    // makes sure files are deleted before updating tracking files
+    if (this.getFlag<boolean>('tracksource')) {
+      await updateTracking({
+        ux: this.ux,
+        result: this.deployResult,
+        tracking: this.tracking,
+        fileResponses,
+      });
+    }
     return result;
   }
 
+  protected async preChecks(): Promise<void> {
```

Comment:
can we move `preChecks` to a base class to keep it DRY?

Reply:
after our discussion, it's more consolidated that it was (always checking source tracking file format).  I'm going to leave this up to Steve to decide how to organize the class hierarchy.

We'd have to move SourceTracking (the class) and `preChecks` all the way up to SourceCommand when so many things that inherit from it won't use it